doc/specs/#148 - Repair Support.md

```diff
@@ -0,0 +1,77 @@
+  - Msix : The repair command will use the built-in repair features of the MSIX installer type. It will make an MSIX API call to register the application package, which will attempt to repair the package.
+    > We can't do the same thing as setting app repair for MSIX app right now because we can't use the private APIs that it uses to do repair operations with winget.
+  - MSStore : The repair command will make an MSStore API  [StartProductInstallAsync](https://learn.microsoft.com/en-us/uwp/api/windows.applicationmodel.store.preview.installcontrol.appinstallmanager.startproductinstallasync?view=winrt-22621) call to repair the application with 'Repair' property of AppInstallOption set to true.
+
+### Repair Feature for Installer Types that require Custom Repair Switch
+  -  Burn/Exe : The custom switch for repair in the YAML manifest file will be used to perform the repair. The repair command will run the installer with the custom switch to fix the application. To have enough flexibility, different options are possible depending on the installer source used for repair. 
+   - Installed Source: If the YAML manifest file has the 'ModifyRepair' switch, the repair command will use the modify command in the ARP 'ModifyPath' registry key with the repair switch through a ShellExecute call.
+   - Remote Source: If the YAML manifest file has the 'InstallerRepair' switch, the repair command will get the matching installer from the remote source and use the repair switch on the downloaded installer through a ShellExecute call.
+      > If neither switch is specified, the repair command will display an error message.
+
+> Note: The initial implementation will not support repair for Nullsoft, Inno, Portables installer types. Based on feedback from the community, we may add the repair feature for these installer types in a future release.
+
+## Manifest Changes
+Addition of `ModifyRepair` property to InstallerSwitch
+- The `ModifyRepair` property is used to define the custom repair option that works with ModifyPath ARP entry for those installer types that have ModifyPath ARP entry and allow repair through Modify flow.
+
+Addition of `InstallerRepair` property to InstallerSwitch
+-  With the `InstallerRepair` switch, the installer that matches the remote source is downloaded and then repaired by using ShellExecute to call the repair switch on the downloaded installer.
+
+## Manifest Validation
+- `ModifyRepair` and `InstallerRepair` switches are only valid for Burn/Exe installer types.
```

Comment:
I agree that a warning is more appropriate than an error for installer types that support custom repair switches, instead of the default experience. However, I wonder if this warning scenario should only apply to limited installer types, such as MSI/Wix and Burn/Exe, that may involve custom repair switch scenarios. For other installer types, such as MSIX/MSStore, Nullsoft, Inno, or portables, showing an error may be more sensible, since the repair action is either to invoke platform APIs or to uninstall and/or reinstall. This way, we can avoid possible manifest inconsistencies.

Please share your thoughts on this.

Reply:
I think that makes sense, especially since there shouldn't be switches which affect the API's behaviors or a reinstall